snapcraft/plugins/v1/colcon.py

```diff
@@ -380,13 +383,17 @@ def _setup_apt_dependencies(self, apt_dependencies):
 
         logger.info("Installing apt dependencies...")
         try:
-            repo.Ubuntu.install_stage_packages(
```

Comment:
I probably would have left the `install_stage_packages` as a (refactored) helper so the ROS stuff didn't need to be touched, but it's done now :D

Reply:
Yeah, I was not aware of how many places this was in, I was thinking of using a tmp thing, but then thought it might be easier to debug in the future that what was unpacked if we can see the debs and not depend on a tmp_dir which I would need to create to solve the decoupling (or pass the list of dl_paths)